docs/core/extensions/localization.md

```diff
@@ -0,0 +1,233 @@
+---
+title: Localization in .NET
+description: Learn how to use the IStringLocalizer and IStringLocalizerFactory implementations with dependency injection in your .NET workloads.
+author: IEvangelist
+ms.author: dapine
+ms.date: 08/06/2021
+---
+
+# Localization in .NET
+
+In this article, you will learn how to use the <xref:Microsoft.Extensions.Localization.IStringLocalizer%601> and <xref:Microsoft.Extensions.Localization.IStringLocalizerFactory> implementations. Localization is the process of translating an application's resources into localized versions for each culture that the application supports.
+
+All of the example source code in this article relies on the [`Microsoft.Extensions.Localization`](https://www.nuget.org/packages/microsoft.extensions.localization) and [`Microsoft.Extensions.Hosting`](https://www.nuget.org/packages/microsoft.extensions.hosting) NuGet packages. For more information on hosting, see [.NET Generic Host](generic-host.md).
+
+## Resource files
+
+The primary mechanism for isolating localizable strings is with **resource files**. A resource file is an XML file with the *.resx* file extension. Resource files are translated prior to the execution of the consuming application &mdash; in other words, they represent translated content at rest. A resource file name most commonly contains a locale identifier, and takes on the following form:
+
+**`{FullTypeName}[.Locale].resx`**
+
+Where:
+
+- The `{FullTypeName}` represents localizable resources for a specific type.
+- The optional `[.Locale]` represents the locale of the of the resource file contents.
+
+### Specifying locales
+
+The locale at a bare minimum should define the language, but it can also define the culture (dialect), and even the country. These segments are commonly delimited by the `-` character. With the added specificity of a culture, the "culture fallback" rules are applied where best matches are prioritized. The locale should map to a well-known language tag, see <xref:System.Globalization.CultureInfo.Name?displayProperty=nameWithType>.
```

Comment:
```suggestion
The locale should define the language, at a bare minimum, but it can also define the culture (dialect), and even the country. These segments are commonly delimited by the `-` character. With the added specificity of a culture, the "culture fallback" rules are applied where best matches are prioritized. The locale should map to a well-known language tag. For more information, see <xref:System.Globalization.CultureInfo.Name?displayProperty=nameWithType>.
```